### PR TITLE
{spring-cloud} Add `spring-cloud create --sku Enterprise`

### PR DESCRIPTION
**Pull request by yuwzho**
---
## Description
This PR adds `az spring-cloud create --sku Enterprise` choice. Since when `sku=Enterprise`, the creation steps are very different with others (later PR will introduce some other parameters which can be used in Enterprise only), this PR adds a router before `custom.py` for create command, routing the command whose `sku==Enterprise` to another impl file. In the new impl file, there will be some steps like creating the top level resource, later creation nested resource for enterprise only. All tiers shares the same step to create top level resource, the new impl file invokes the top level resource creation impl from `custom.py` as well.

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring-cloud

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def load_arguments(self, _):
     # https://dev.azure.com/msazure/AzureDMSS/_workitems/edit/11002857/
     with self.argument_context('spring-cloud create') as c:
         c.argument('location', arg_type=get_location_type(self.cli_ctx), validator=validate_location)
+        c.argument('sku', arg_type=get_enum_type(['Basic', 'Standard', 'Enterprise']), validator=validate_sku, default='Standard', help='Name of SKU. Enterprise is still in Preview.')
```
Comment (wangzelin007) [inline src/spring-cloud/azext_spring_cloud/_params.py:43]:
```suggestion
sku_type = CLIArgumentType(arg_type=get_enum_type(['Basic', 'Standard', 'Enterprise']), validator=validate_sku, help='Name of SKU. Enterprise is still in Preview.')
        c.argument('sku', arg_type=sku_type, default='Standard')
```

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 2]
```diff
@@ -72,6 +73,7 @@ def load_arguments(self, _):
                    validator=validate_tracing_parameters_asc_create)
 
     with self.argument_context('spring-cloud update') as c:
+        c.argument('sku', arg_type=get_enum_type(['Basic', 'Standard', 'Enterprise']), validator=validate_sku, help='Name of SKU. Enterprise is still in Preview.')
```
Comment (wangzelin007) [inline src/spring-cloud/azext_spring_cloud/_params.py:76]:
```suggestion
        c.argument('sku', arg_type=sku_type)
```

**Review by wangzelin007 — CHANGES_REQUESTED** [review]
Please write the description of changes into HISTORY.rst and update setup.py to release the new extension version.

File: src/spring-cloud/azext_spring_cloud/_params.py (hunk 1 above)
Comment (yuwzho) [reply src/spring-cloud/azext_spring_cloud/_params.py]: Updated

File: src/spring-cloud/azext_spring_cloud/_params.py (hunk 2 above)
Comment (yuwzho) [reply src/spring-cloud/azext_spring_cloud/_params.py]: Updated

**Comment by yuwzho** [comment]
> Please write the description of changes into HISTORY.rst and update setup.py to release the new extension version.

No, we will announce the public preview on Jan. 17th, for the public preview, we have a lot of new features contributed to CLI. Will update the release notes and version together after all changes merged.

**Review by wangzelin007 — APPROVED** [review]
LGTM

**Comment by yuwzho** [comment]
#4294

**Pull request merged**